src/Adapter/Product/ProductPersister.php

```diff
@@ -54,43 +52,22 @@ public function __construct(
         $this->productValidator = $productValidator;
     }
 
+    //@todo: add(Product $product): ProductId
+
     /**
      * @param Product $product
      * @param array $propertiesToUpdate
      * @param int $errorCode
      *
      * @throws CoreException
      */
-    public function update(Product $product, array $propertiesToUpdate, int $errorCode): void
+    public function partialUpdate(Product $product, array $propertiesToUpdate, int $errorCode): void
     {
         $this->fillProperties($product, $propertiesToUpdate);
         $this->productValidator->validate($product);
         $this->updateObjectModel($product, CannotUpdateProductException::class, $errorCode);
     }
 
-    /**
-     * @param Product $product
-     */
-    public function refreshProductCustomizabilityProperties(Product $product): void
-    {
-        if ($product->hasActivatedRequiredCustomizableFields()) {
-            $customizable = ProductCustomizabilitySettings::REQUIRES_CUSTOMIZATION;
-        } elseif (!empty($product->getNonDeletedCustomizationFieldIds())) {
-            $customizable = ProductCustomizabilitySettings::ALLOWS_CUSTOMIZATION;
-        } else {
-            $customizable = ProductCustomizabilitySettings::NOT_CUSTOMIZABLE;
-        }
-
-        $this->update(
-            $product,
-            [
-                'customizable' => $customizable,
-                'text_fields' => $product->countCustomizationFields(CustomizationFieldType::TYPE_TEXT),
-                'uploadable_files' => $product->countCustomizationFields(CustomizationFieldType::TYPE_FILE),
-            ], CannotUpdateProductException::FAILED_UPDATE_CUSTOMIZATION_FIELDS
-        );
-    }
-
     /**
      * @param Product $product
      * @param array $propertiesToUpdate
```

Comment:
While you're cleaning this one, can you create the sub function `fillCustomizationFields` please? which will be called by `fillProperties`